Backend/Tournament_User/TournamentUserMVSC/models/tournamentModel.js

```diff
@@ -0,0 +1,208 @@
+        db.query(query, [UUID, tournamentID, elo], (err, results) => {
+            if (err) {
+                return reject(err);  // Return the error to the service layer
+            }
+            resolve(results);  // Return the success result to the service layer
+        });
+    });
+};
+
+/**
+ * Allows a user to quit a tournament they have signed up for.
+ * @param {string} UUID - The UUID of the user.
+ * @param {string} tournamentID - The ID of the tournament.
+ * @returns {Promise<Object>} A promise that resolves when the user successfully quits the tournament.
+ * @throws {Error} If the database query fails.
+ */
+exports.quitTournament = (UUID, tournamentID) => {
+    return new Promise((resolve, reject) => {
+        const query = 'CALL UserQuit(?, ?)';  // Stored procedure to remove a user from a tournament
+        db.query(query, [UUID, tournamentID], (err, results) => {
+            if (err) {
+                return reject(err);  // Return the error to the service layer
+            }
+            resolve(results);  // Return the success result to the service layer
+        });
+    });
+};
+
+/**
+ * Retrieves a list of upcoming tournaments.
+ * @returns {Promise<Array>} A promise that resolves to an array of upcoming tournaments.
+ * @throws {Error} If the database query fails.
+ */
+exports.getUpcomingTournaments = () => {
+    return new Promise((resolve, reject) => {
+        const query = 'CALL GetUpComingTournament()';  // Stored procedure to get upcoming tournaments
+        db.query(query, (err, results) => {
+            if (err) {
+                return reject(err);  // Return the error to the service layer
+            }
+            resolve(results);  // Return the upcoming tournaments to the service layer
+        });
+    });
+};
+
+/**
+ * Retrieves a list of tournaments that are currently in progress.
+ * @returns {Promise<Array>} A promise that resolves to an array of in-progress tournaments.
+ * @throws {Error} If the database query fails.
+ */
+exports.getInProgressTournaments = () => {
+    return new Promise((resolve, reject) => {
+        const query = 'CALL GetInProgressTournament()';  // Stored procedure to get in-progress tournaments
+        db.query(query, (err, results) => {
+            if (err) {
+                return reject(err);  // Return the error to the service layer
+            }
+            resolve(results);  // Return the in-progress tournaments to the service layer
+        });
+    });
+};
+
+/**
+ * Retrieves the ranking and win count of a user in a specific tournament.
+ * @param {string} tournamentID - The ID of the tournament.
+ * @param {string} UUID - The UUID of the user.
+ * @returns {Promise<Object>} A promise that resolves to the user's tournament game rank and win count.
+ * @throws {Error} If the database query fails.
+ */
+exports.getUserTournamentGameRank = (tournamentID, UUID) => {
```

Comment:
i think u should separate the service layers ie one for tournaments and one for users cos this file is pretty long